modules/generative-openai/clients/openai.go

```diff
@@ -32,33 +32,37 @@ import (
 
 var compile, _ = regexp.Compile(`{([\w\s]*?)}`)
 
-func buildUrlFn(isLegacy bool, resourceName, deploymentID string) (string, error) {
+func buildUrlFn(isLegacy bool, resourceName, deploymentID, baseURL string) (string, error) {
+	if baseURL == "" {
+		baseURL = "https://api.openai.com"
+	}
```

Comment:
this part won't be needed since the `baseURL` value will come from `class_settings` which if not overriden will be always set to `"https://api.openai.com"`